leptos_macro/src/view/component_builder.rs

```diff
@@ -267,11 +277,15 @@ pub(crate) fn component_to_tokens(
             ::leptos::component::component_view(
                 #[allow(clippy::needless_borrows_for_generic_args)]
                 &#name,
-                ::leptos::component::component_props_builder(&#name #generics)
-                    #(#props)*
-                    #(#slots)*
-                    #children
-                    .build()
+                {
+                    let mut props = ::leptos::component::component_props_builder(&#name #generics)
```

Comment:
This pattern has the potential to generate `unused_mut` and `clippy::let_and_return` warnings. In my current version, I don't see clippy applying these to the macro version but if I expand the macro code, it does apply those warnings, so it might start applying them in the future. It would be good to add these to the list of allowed lints above.

Reply:
Added!